feat: update read me (#353)
PeriodPals/periodpals: README.md

# Welcome to PeriodPals!

> Our app is designed to ensure that no one ever has to worry about being without menstrual products when they need them most. Quickly send an alert detailing your needs, whether it's a tampon or a pad, along with the urgency, location and a personal message to express yourself. Users near you will receive a notification on their phone, which notifies them that your alert has been posted. They can also see all the active alerts in a list, filter them by location, product or urgency and respond to them if they can help. If someone can help you, they accept your alert and establish a chat with you to communicate easily about where to meet to give you the product. 

Reviewer: ```suggestion
Our app is designed to ensure that no one ever has to worry about being without menstrual products when they need them most. Quickly send alerts detailing your needs, whether it's a tampon or a pad, along with the urgency, location and a personal message to express yourself. Users nearby will receive a push notification on their phone, which inform them that your alert has been posted. They can also see all the active alerts in a list, filter them by location, product or urgency and respond to them if they can help. If someone can help you, they accept your alert and connect with you through a chat to communicate easily about where to meet to give you the product. 
```

> The active alerts also appear on an interactive map, enabling community members nearby to respond and connect with you through one-on-one chat. 

Reviewer: ```suggestion
The active alerts also appear on an interactive map, enabling community members nearby to visualize your location and connect with you more easily. 
```

> _It’s about getting you the help you need, exactly when you need it._

Reviewer: I really like this!

## Key Features 
### Authentication and Profile

> Users can create a profile using their email and choose a secure password. Upon creating their account, they can customize their profile by providing their name, date of birth (users must be over 16 years old), and a personal description. They can also upload a profile picture. Once registered, users can easily log in across different devices using their email and password. 

Reviewer: ```suggestion
Users can create a profile using their email and choose a secure password. Upon creating their account, they can customize their profile by providing their name, date of birth (users must be over 16 years old), and a personal description. They can also upload a profile picture. Once registered, users can easily log in across different devices using their credentials. 
```

> Users can view their profile and edit all fields. From the profile screen, they can also access settings.

Reviewer: ```suggestion
Users can view their profile and edit all fields. From the profile screen, they can also access their settings.
```

### Creating an Alert
Users can create an alert by filling out a form with all necessary information (what is needed, their location, etc.). Once the alert is created, it can be viewed in the "Lists" under the "My Alerts" tab, where users can edit the alert, delete it, or mark it as solved when they have received the help they needed.

### Responding to an Alert
Users will receive notifications on their phones when another user within a specified radius of their location sends an alert. The preferred radius is set upon account creation and can be modified from the profile settings.

Alerts can be viewed in list form or on a map. In the "Lists" under the "Pal's Alerts" tab, users can see alerts posted by others, with icons indicating the product type and urgency, as well as location, time posted, and other necessary information. Clicking on an alert reveals an option to accept it, which initiates a chat with the person who posted the alert.

> On the map, alerts are marked with icons, alongside the user's current location, enabling spatial visualization of alerts that can be accepted. Clicking on an alert icon displays the alert's information and an "Accept" button.

Reviewer 2: Maybe mention too that if the user clicks on one of their alerts, they can resolve it or edit it.

TBD: This feature is not yet finalized as users who post an alert currently do not receive a notification when another user responds to their alert.

> ### Setting a Timer to track Tampon use

Reviewer 2: Track should be capitalized.

> A timer is available for users to track the duration of tampon use, addressing a critical issue for those who menstruate: the risk of forgetting to remove their tampons on time, which can lead to serious health issues such as Toxic Shock Syndrome. The timer starts a countdown from 6 hours and sends reminders to the user when it's time to change the tampon.

Reviewer: In here, can you mention that the 6h is the medically recommended time?

> ## Links

Reviewer: I think this section could be renamed "Implementation details" or something of the kind, so that inside we can put a little description of what API's we use, how we ensure our users' data is secure (most importantly their location), etc. It would make sense to have the architecture diagram as an image, and to add the link our architecture diagram, figma, powersync instance etc.

Author: I don't think have the image for the architecture diagram is a good idea, mainly because of the quality of the image.

Author: Forge it, quality is fine, I will add the image!
